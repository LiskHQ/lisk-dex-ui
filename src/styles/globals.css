:root {
  font-size: 16px;
}

html,
body,
#__next {
  padding: 0 !important;
  margin: 0;
  height: 100%;
}

a {
  color: inherit;
  text-decoration: none;
}

* {
  box-sizing: border-box;
}

a,
.btn,
button,
span,
p,
i,
input,
select,
textarea,
li,
img,
svg path,
*::after,
*::before,
.transition-3,
h1,
h2,
h3,
h4,
h5,
h6 {
  transition: all 0.3s ease-out 0s;
}

span,
p,
a,
li.MuiMenuItem-root,
label,
label.MuiFormLabel-root,
span.MuiTypography-root,
p.MuiTypography-root,
button.MuiButton-root,
button.MuiButtonBase-root,
.MuiInput-root,
.MuiTableCell-root {
  font-family: 'Roboto', sans-serif;
}

h1,
h2,
h3,
h4,
h5,
h6,
h1.MuiTypography-root,
h2.MuiTypography-root,
h3.MuiTypography-root,
h4.MuiTypography-root,
h5.MuiTypography-root,
h6.MuiTypography-root {
  font-family: 'Roboto', sans-serif;
}