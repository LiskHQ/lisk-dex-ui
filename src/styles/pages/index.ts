import { GovernanceStyle } from "./governance/index.style";

export {
  GovernanceStyle
}