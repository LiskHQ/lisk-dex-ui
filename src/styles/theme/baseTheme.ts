export const baseTheme = {
  typography: {
    h1: {
      fontSize: "2.5rem",
      fontWeight: 600,
    },
    h2: {
      fontSize: "2rem",
      fontWeight: 600,
    },
    h3: {
      fontSize: "1.75rem",
      lineHeight: "2rem",
      fontWeight: 500,
    },
    h4: {
      fontSize: "1.25rem",
      fontWeight: 600,
    },
    h5: {
      fontSize: "1rem",
      lineHeight: "1.25rem",
      fontWeight: 500,
    },
    h6: {
      fontSize: "0.875rem",
      lineHeight: "1rem",
      fontWeight: 500,
    },
    body1: {
      fontSize: "1rem",
      lineHeight: "1.375rem",
      fontWeight: 500,
    },
    body2: {
      fontSize: "0.875rem",
      lineHeight: "1.125rem",
      fontWeight: 400,
    },
    caption: {
      fontSize: "0.875rem",
      lineHeight: "1rem",
      fontWeight: 500,
    },
    subtitle1: {
      fontSize: "1.75rem",
      lineHeight: "2rem",
      fontWeight: 500,
    },
    subtitle2: {
      fontSize: "1.25rem",
      lineHeight: "1.5rem",
      fontWeight: 400,
    }
  },
  icons: {
    sm: "1rem",
    md: "1.25rem",
    lg: "1.5rem",
  },
  breakpoints: {
    values: {
      xs: 0,
      sm: 576,
      md: 768,
      lg: 1024,
      xl: 1440,
    },
  },
  shadows: {
    sm: "0px 1px 3px rgba(0, 0, 0, 0.1)"
  }
}