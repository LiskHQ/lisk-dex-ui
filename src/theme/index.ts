export { default as theme } from "./theme";
export { default as variables } from "./variables";
