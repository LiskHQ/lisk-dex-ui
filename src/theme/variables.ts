const variables = {
  black: "#000000",
  white: "#ffffff",
  primary: {
    superPurple: "#6953F4",
    lighter: "#5845CC",
    light: "#4738A6",
    main: "#362B7F",
    dark: "#261E59",
    superDark: "#161133",
    contrastText: "#fff",
    stone: "#323d47",
    darkSilver: "#757575",
  },
  secondary: {
    superPurple: "#F5981B",
    lighter: "#CC7E16",
    light: "#A66712",
    main: "#7F4F0E",
    dark: "#59370A",
    superDark: "#332006",
  },
  opacities: {
    superPurple: "#F5981B",
    lighter: "#CC7E16",
    light: "#A66712",
    main: "#7F4F0E",
    dark: "#59370A",
    superDark: "#332006",
    tokenBackground: "#F2F5F9",
  },
  success: {
    superPurple: "#E3F2E0",
    lighter: "#8DC881",
    light: "#459A33",
  },
  warning: {
    superPurple: "#F8F3D8",
    lighter: "#CC7E16",
    light: "#C9A81C",
  },
  information: {
    superPurple: "#F5981B",
    lighter: "#CC7E16",
    light: "#A66712",
  },
  error: {
    superPurple: "#F5981B",
    lighter: "#CC7E16",
    light: "#A66712",
  },
  background: "#E1E3E6",
  lineColor: "#e0e2f1",
};

export default variables;
