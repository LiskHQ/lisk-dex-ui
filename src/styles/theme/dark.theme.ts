import { createTheme } from '@mui/material/styles';
import { baseTheme } from './baseTheme';

const themeColors = {
  text: {
    primary: '#F2F2F2',
    secondary: '#F2F2F2',
    paragraph: '#CDCDCD',
    body: '#F2F2F2',
    heading: '#4738A6',
    button: '#FFFFFF',
    placeholder: '#A4ABB7',
  },
  lightcurve: {
    0: '#6953F4',
    10: '#5845CC',
    20: '#4738A6',
    40: '#362B7F',
    60: '#261E59',
    80: '#E6E6E6',
  },
  secondary: {
    0: '#F5981B'
  },
  primary: {
    85: '#7E6CF4',
    70: '#4F39B2',
    60: '#271B5B',
    40: '#2D2166',
    20: '#322573',
    10: '#4F39B2',
    5: '#322573',
    2.5: '#25195E',
    1: '#F4F8F9',
    0: '#FFFFFF',
  },
  opacities: {
    5: '#333333',
    10: '#3D3D3D',
    20: '#E6E6E6',
    40: '#7E7E7E',
    80: '#D4D4D4',
  },
  border: {
    primary: '#433199',
  },
  bg: {
    primary: 'linear-gradient(90deg, #140C42 0%, #211953 52.88%, #110A3A 101.53%, #140C42 101.53%)',
    secondary: '#22184C',
    helper: '#6953F4',
    proposalItem: 'linear-gradient(180deg, #2E2567 0%, rgba(37, 26, 103, 0.5) 143.01%);',
    banner: 'linear-gradient(180deg, #150556 0%, rgba(25, 9, 98, 0.9) 100%)',
    walletAddress: 'linear-gradient(180deg, #372589 0%, rgba(27, 19, 68, 0.8) 100%)',
    slider: '#322573',
  },
  success: {
    primary: '#E3F2E0',
    second: '#8DC881',
  },
  warning: {
    primary: '#F8F3D8',
    second: '#E2CA64',
  },
  info: {
    primary: '#2EA3DD',
    second: '#C0ECE8',
  },
  error: {
    primary: '#F6E4E6',
    second: '#FB8B96',
  },
} as const;

const lightTheme = createTheme({
  ...themeColors,
  ...baseTheme,
  palette: {
    mode: 'dark',
  },
})

export default lightTheme
