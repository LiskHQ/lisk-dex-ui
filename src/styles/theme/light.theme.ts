import { createTheme } from '@mui/material/styles';
import { baseTheme } from './baseTheme';

const themeColors = {
  text: {
    primary: '#282828',
    secondary: '#515661',
    paragraph: '#6B7280',
    body: '#6B7280',
    heading: '#4738A6',
    button: '#FFFFFF',
    placeholder: '#6B7280',
  },
  lightcurve: {
    0: '#6953F4',
    10: '#A196F6',
    20: '#4738A6',
    40: '#362B7F',
    60: '#261E59',
    80: '#161133',
  },
  primary: {
    85: '#7E6CF4',
    60: '#A196F6',
    40: '#E4EBF9',
    20: '#D9D8F8',
    10: '#E7E9F9',
    5: '#EEF0F8',
    2.5: '#F2F5F9',
    1: '#F4F8F9',
    0: '#6953F4'
  },
  opacities: {
    5: '#333333',
    10: '#3D3D3D',
    20: '#535353',
    40: '#7E7E7E',
    80: '#D4D4D4',
  },
  border: {
    primary: '#E6E6E6',
  },
  bg: {
    primary: '#FBFBFB',
    secondary: '#FFFFFF',
    helper: '#22184D',
    banner: 'linear-gradient(180deg, #F3F3FD 0%, rgba(238, 238, 249, 0.9) 100%)',
    walletAddress: 'linear-gradient(180deg, #EEEFF7 0%, rgba(232, 234, 245, 0.8) 100%);',
    slider: '#F2F2F2',
  },
  success: {
    primary: '#E3F2E0',
    second: '#8DC881',
  },
  warning: {
    primary: '#F8F3D8',
    second: '#E2CA64',
  },
  info: {
    primary: '#EBF7F6',
    second: '#C0ECE8',
  },
  error: {
    primary: '#F6E4E6',
    second: '#FB8B96',
  },
} as const;

const lightTheme = createTheme({
  ...themeColors,
  ...baseTheme,
  palette: {
    mode: 'light'
  },
})

export default lightTheme
