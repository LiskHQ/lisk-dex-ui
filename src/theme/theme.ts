import {
  createTheme,
  responsiveFontSizes,
  ThemeOptions,
} from "@mui/material/styles";

import variables from "./variables";

const options: ThemeOptions = {
  palette: {
    primary: variables.primary,
    secondary: variables.secondary,
  },
  typography: {
    h1: {
      fontSize: "2.5rem",
      fontWeight: 600,
    },
    h2: {
      fontSize: "2rem",
      fontWeight: 400,
    },
    h3: {
      fontSize: "1.75rem",
      fontWeight: 400,
    },
    h4: {
      fontSize: "1.25rem",
      fontWeight: 400,
    },
    h5: {
      fontSize: "1rem",
      fontWeight: 400,
    },
    h6: {
      fontSize: "0.75rem",
      fontWeight: 400,
    },
  },
  breakpoints: {
    values: {
      xs: 0,
      sm: 576,
      md: 768,
      lg: 1024,
      xl: 1440,
      xxl: 1950,
      xxxl: 2300,
    },
  },
};

const theme = responsiveFontSizes(createTheme(options));
export default theme;
