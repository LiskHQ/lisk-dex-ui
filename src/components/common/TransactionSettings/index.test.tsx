import React from 'react';
import { ThemeProvider } from '@mui/material';
import { fireEvent, render } from '@testing-library/react';
import { TransactionSettings, ITransactionSettingsProps } from './index';
import { lightTheme } from 'styles/theme';

function renderComponent(props: ITransactionSettingsProps) {
  return render(
    <ThemeProvider theme={lightTheme}>
      <TransactionSettings {...props} />
    </ThemeProvider>
  );
}

describe('TransactionSettings component', () => {
  const mockProps: ITransactionSettingsProps = {
    splipageTolerance: 0.5,
    transactionDeadline: 100,
    onChangeSplipageTolerance: jest.fn(),
    onChangeTransactionDeadline: jest.fn(),
  };

  it('checks if the component matches the snapshot', () => {
    const { container } = renderComponent(mockProps);
    expect(container).toMatchSnapshot();
  });

  it('test splipageTolerance change', () => {
    const { getByTestId } = renderComponent(mockProps);
    const button = getByTestId('splipageTolerance-test-1');
    fireEvent.click(button);
    expect(mockProps.onChangeSplipageTolerance).toBeCalledWith(1);
  });


  it('test transactionDeadline change', () => {
    const { getByTestId } = renderComponent(mockProps);
    const input = getByTestId('transaction-deadline-test').querySelector('.MuiInputBase-input');
    fireEvent.change(input, { target: { value: 1000 } });
    expect(mockProps.onChangeTransactionDeadline).toBeCalledWith(1000);
  });
});
