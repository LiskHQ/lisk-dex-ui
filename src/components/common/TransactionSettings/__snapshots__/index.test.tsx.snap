// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionSettings component checks if the component matches the snapshot 1`] = `
.emotion-0 .transactioin-settings-slipage-tolerance {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 2rem;
}

.emotion-0 .transactioin-settings-slipage-tolerance svg {
  margin-left: 0.25rem;
}

.emotion-0 .transaction-settings-slipage-tolerance-switcher {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  margin-top: 0.75rem;
}

.emotion-0 .transaction-settings-slipage-tolerance-switcher .switcher-box {
  border-radius: 0.25rem;
  border: 0.75px solid #E6E6E6;
  padding: 0.75rem 1rem;
  width: 4.375rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  cursor: pointer;
}

.emotion-0 .transaction-settings-slipage-tolerance-switcher .switcher-box.selected {
  background: #6953F4;
  color: #FFFFFF;
  border: 0.75px solid #4738A6;
}

.emotion-0 .transaction-settings-slipage-tolerance-switcher .slipage-tolerance-input {
  width: 11.5rem;
}

.emotion-0 .transaction-settings-slipage-tolerance-switcher .slipage-tolerance-input .MuiInputBase-input {
  text-align: right;
}

.emotion-0 .transaction-deadline-label {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 2rem;
}

.emotion-0 .transaction-deadline-label svg {
  margin-left: 0.25rem;
}

.emotion-0 .transaction-deadline-input {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 0.75rem;
}

.emotion-0 .transaction-deadline-input .MuiFormControl-root {
  width: 5.625rem;
  height: 2.75rem;
  margin-right: 0.75rem;
}

.emotion-1 {
  margin: 0;
  font-size: 1rem;
  line-height: 1.375rem;
  font-weight: 500;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  letter-spacing: 0.00938em;
}

.emotion-9 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  min-width: 0;
  padding: 0;
  margin: 0;
  border: 0;
  vertical-align: top;
  position: relative;
}

.emotion-9 .MuiInputLabel-root {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: unset;
  color: #282828!important;
  font-size: 0.875rem;
  line-height: 1.124rem;
  font-weight: 400;
  -webkit-transform: none;
  -moz-transform: none;
  -ms-transform: none;
  transform: none;
  -webkit-transform: none;
  -ms-transform: none;
  margin-bottom: 0.5rem;
}

.emotion-9 .MuiInputLabel-root svg {
  cursor: pointer;
  margin-left: 0.25rem;
}

.emotion-9 .MuiInputLabel-root svg path {
  stroke: #6B7280!important;
}

.emotion-9 .MuiInputBase-root {
  padding: 0.75rem 1rem;
  border-radius: 0.25rem;
  color: #282828;
  border: 0.5px solid #E6E6E6;
  background: unset;
  -webkit-animation-duration: 1s!important;
  animation-duration: 1s!important;
  font-size: 1rem;
  line-height: 1.625rem;
}

.emotion-9 .MuiInputBase-root input {
  padding: 0;
}

.emotion-9 .MuiInputBase-root fieldset {
  border: none;
}

.emotion-9 .MuiInputBase-root:before {
  content: none;
}

.emotion-9 .MuiInputBase-root:after {
  content: none;
}

.emotion-9 .MuiInputBase-root:hover {
  border: 1px solid #A196F6;
}

.emotion-9 .MuiInputBase-root:focus-within {
  border: 1px solid #6953F4;
}

.emotion-9 .MuiFormHelperText-root {
  color: #FF4557;
  margin: 0.5rem 0 0 0;
}

.emotion-9 .MuiFormHelperText-root p {
  text-align: right;
}

.emotion-9 .input-length-counter {
  position: absolute;
  bottom: -1.5rem;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: end;
  justify-content: end;
  color: #6B7280;
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
}

.emotion-9 input::-webkit-outer-spin-button,
.emotion-9 input::-webkit-inner-spin-button {
  -webkit-appearance: none;
  display: none;
}

.emotion-10 {
  font-size: 1rem;
  line-height: 1.4375em;
  font-weight: 500;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  letter-spacing: 0.00938em;
  color: rgba(0, 0, 0, 0.87);
  box-sizing: border-box;
  position: relative;
  cursor: text;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-10.Mui-disabled {
  color: rgba(0, 0, 0, 0.38);
  cursor: default;
}

.emotion-11 {
  font: inherit;
  letter-spacing: inherit;
  color: currentColor;
  padding: 4px 0 5px;
  border: 0;
  box-sizing: content-box;
  background: none;
  height: 1.4375em;
  margin: 0;
  -webkit-tap-highlight-color: transparent;
  display: block;
  min-width: 0;
  width: 100%;
  -webkit-animation-name: mui-auto-fill-cancel;
  animation-name: mui-auto-fill-cancel;
  -webkit-animation-duration: 10ms;
  animation-duration: 10ms;
}

.emotion-11::-webkit-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-11::-moz-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-11:-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-11::-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-11:focus {
  outline: 0;
}

.emotion-11:invalid {
  box-shadow: none;
}

.emotion-11::-webkit-search-decoration {
  -webkit-appearance: none;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-11::-webkit-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-11::-moz-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-11:-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-11::-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-11:focus::-webkit-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-11:focus::-moz-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-11:focus:-ms-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-11:focus::-ms-input-placeholder {
  opacity: 0.42;
}

.emotion-11.Mui-disabled {
  opacity: 1;
  -webkit-text-fill-color: rgba(0, 0, 0, 0.38);
}

.emotion-11:-webkit-autofill {
  -webkit-animation-duration: 5000s;
  animation-duration: 5000s;
  -webkit-animation-name: mui-auto-fill;
  animation-name: mui-auto-fill;
}

.emotion-11::-webkit-input-placeholder {
  opacity: 1;
  color: #6B7280;
}

.emotion-11::-moz-placeholder {
  opacity: 1;
  color: #6B7280;
}

.emotion-11:-ms-input-placeholder {
  opacity: 1;
  color: #6B7280;
}

.emotion-11::placeholder {
  opacity: 1;
  color: #6B7280;
}

.emotion-17 {
  margin: 0;
  font-size: 0.875rem;
  line-height: 1.125rem;
  font-weight: 400;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  letter-spacing: 0.01071em;
}

<div>
  <div
    class="emotion-0"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 transactioin-settings-slipage-tolerance emotion-1"
    >
      Slippage Tolerance 
      <svg
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          clip-path="url(#clip0_5_93111)"
        >
          <path
            d="M7.9987 14.6673C11.6806 14.6673 14.6654 11.6825 14.6654 8.00065C14.6654 4.31875 11.6806 1.33398 7.9987 1.33398C4.3168 1.33398 1.33203 4.31875 1.33203 8.00065C1.33203 11.6825 4.3168 14.6673 7.9987 14.6673Z"
            stroke="#6B7280"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.3"
          />
          <path
            d="M6.05859 6.00038C6.21533 5.55482 6.5247 5.17912 6.9319 4.9398C7.3391 4.70049 7.81786 4.61301 8.28338 4.69285C8.7489 4.7727 9.17114 5.01473 9.47531 5.37606C9.77949 5.7374 9.94596 6.19473 9.94526 6.66705C9.94526 8.00038 7.94526 8.66705 7.94526 8.66705"
            stroke="#6B7280"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.3"
          />
          <path
            d="M8 11.334H8.00708"
            stroke="#6B7280"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.3"
          />
        </g>
        <defs>
          <clippath
            id="clip0_5_93111"
          >
            <rect
              fill="white"
              height="16"
              width="16"
            />
          </clippath>
        </defs>
      </svg>
    </p>
    <div
      class="transaction-settings-slipage-tolerance-switcher MuiBox-root emotion-2"
    >
      <div
        class="switcher-box MuiBox-root emotion-2"
        data-testid="splipageTolerance-test-0.1"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-1"
        >
          0.1
          %
        </p>
      </div>
      <div
        class="switcher-box selected MuiBox-root emotion-2"
        data-testid="splipageTolerance-test-0.5"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-1"
        >
          0.5
          %
        </p>
      </div>
      <div
        class="switcher-box MuiBox-root emotion-2"
        data-testid="splipageTolerance-test-1"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-1"
        >
          1
          %
        </p>
      </div>
      <div
        class="MuiFormControl-root slipage-tolerance-input emotion-9"
      >
        <div
          class="MuiInputBase-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl emotion-10"
          data-testid="input-component-test"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input Mui-disabled emotion-11"
            disabled=""
            type="text"
            value="0.75%"
          />
        </div>
      </div>
    </div>
    <p
      class="MuiTypography-root MuiTypography-body1 transaction-deadline-label emotion-1"
    >
      Transaction deadline 
      <svg
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          clip-path="url(#clip0_5_93111)"
        >
          <path
            d="M7.9987 14.6673C11.6806 14.6673 14.6654 11.6825 14.6654 8.00065C14.6654 4.31875 11.6806 1.33398 7.9987 1.33398C4.3168 1.33398 1.33203 4.31875 1.33203 8.00065C1.33203 11.6825 4.3168 14.6673 7.9987 14.6673Z"
            stroke="#6B7280"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.3"
          />
          <path
            d="M6.05859 6.00038C6.21533 5.55482 6.5247 5.17912 6.9319 4.9398C7.3391 4.70049 7.81786 4.61301 8.28338 4.69285C8.7489 4.7727 9.17114 5.01473 9.47531 5.37606C9.77949 5.7374 9.94596 6.19473 9.94526 6.66705C9.94526 8.00038 7.94526 8.66705 7.94526 8.66705"
            stroke="#6B7280"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.3"
          />
          <path
            d="M8 11.334H8.00708"
            stroke="#6B7280"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.3"
          />
        </g>
        <defs>
          <clippath
            id="clip0_5_93111"
          >
            <rect
              fill="white"
              height="16"
              width="16"
            />
          </clippath>
        </defs>
      </svg>
    </p>
    <div
      class="transaction-deadline-input MuiBox-root emotion-2"
    >
      <div
        class="MuiFormControl-root emotion-9"
      >
        <div
          class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl emotion-10"
          data-testid="transaction-deadline-test"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input emotion-11"
            type="number"
            value="100"
          />
        </div>
      </div>
      <p
        class="MuiTypography-root MuiTypography-body2 emotion-17"
      >
        minutes
      </p>
    </div>
  </div>
</div>
`;
